_examples/ThisExtensionFunc.md

```diff
@@ -0,0 +1,62 @@
+---
+title: this in extension functions
+---
+    
+<div class="sample" markdown="1">
+
+Normally, when you create a class, by default it inherits from *Any*, the Kotlin default supertype which declares toString(), hashCode() and equals().
+In other languages like Java, if you want to extend a class behaviour, you create a *derived* class like this:
+
+```kotlin
+open class Human {                     //1
+    init {
+        println("cogito, ergo sum!")   //2
+    }
+}
+
+class Developer: Human()               //3
+
+fun main(args: Array<String>){
+    val person = Developer()           //4 
+}
+
+```
+
+1. Class that allows to be inheritable. Note: Any is inherted by default
+2. Initializer block, with some code
+3. Derived class definition
+4. Instance creation, with inherited behaviour  
+  
+But in Kotlin, with **extension functions** you have another way to extend class functionalities: just declare a new *fun* on the base class, no need to define another (derived) class and generate its instance.
```

Comment:
But in Kotlin, with **extension functions** you have another way to extend class functionalities: simply declare a new *fun* on the base class, there is no need to define another (derived) class and generate its instance.